sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/e-cert-cancellation-response/_tests_/ecert-cancellation-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,425 @@
+          db.dataSource,
+          undefined,
+          {
+            offeringIntensity: OfferingIntensity.fullTime,
+            applicationStatus: ApplicationStatus.Completed,
+            firstDisbursementInitialValues: {
+              coeStatus: COEStatus.completed,
+              disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+              documentNumber: firstDocumentNumber,
+            },
+          },
+        );
+        const secondApplicationPromise = saveFakeApplicationDisbursements(
+          db.dataSource,
+          undefined,
+          {
+            offeringIntensity: OfferingIntensity.partTime,
+            applicationStatus: ApplicationStatus.Completed,
+            firstDisbursementInitialValues: {
+              coeStatus: COEStatus.completed,
+              disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+              documentNumber: secondDocumentNumber,
+            },
+          },
+        );
+        const [firstApplication, secondApplication] = await Promise.all([
+          firstApplicationPromise,
+          secondApplicationPromise,
+        ]);
+        const [firstDisbursement] =
+          firstApplication.currentAssessment.disbursementSchedules;
+        const [secondDisbursement] =
+          secondApplication.currentAssessment.disbursementSchedules;
+        // Create fake deducted overawards for the first disbursement.
+        const deductedOveraward = createFakeDisbursementOveraward({
+          student: firstApplication.student,
+          studentAssessment: firstApplication.currentAssessment,
+          disbursementSchedule: firstDisbursement,
+        });
+        deductedOveraward.disbursementValueCode = "CSLF";
+        deductedOveraward.overawardValue = 500;
+        deductedOveraward.originType =
+          DisbursementOverawardOriginType.AwardDeducted;
+        await db.disbursementOveraward.save(deductedOveraward);
+        mockDownloadFiles(sftpClientMock, [
+          FULL_TIME_CANCELLATION_RESPONSE_FILE,
+        ]);
+        // Queued job.
+        const mockedJob = mockBullJob<void>();
+
+        // Act
+        const result = await processor.processQueue(mockedJob.job);
+
+        // Assert
+        expect(result).toStrictEqual([
+          "Process finalized with success.",
+          "Received cancellation files: 1.",
+        ]);
+        const downloadedFile = path.join(
+          process.env.ESDC_RESPONSE_FOLDER,
+          FULL_TIME_CANCELLATION_RESPONSE_FILE,
+        );
+        // Check for the log messages.
+        expect(
+          mockedJob.containLogMessages([
+            "Received 1 e-cert cancellation response file(s) to process.",
+            `The downloaded file ${downloadedFile} contains 2 detail records.`,
+            `E-Cert with document number ${firstDocumentNumber} has been cancelled.`,
+            `E-Cert with document number ${firstDocumentNumber} has been cancelled.`,
```

Comment:
As a suggestion, should the overawards have some logs added?
Something like below.
```
CSLF award deducted due to e-cert rejection.
```